nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerASMDump.java

```diff
@@ -70,6 +70,14 @@
             mv.visitMethodInsn(INVOKESTATIC, "edu/illinois/nondex/shuffling/ControlNondeterminism", "shuffle", "(Ljava/util/List;)Ljava/util/List;", false);
             mv.visitVarInsn(ASTORE, 3);
             mv.visitVarInsn(ALOAD, 0);
+            mv.visitFieldInsn(GETFIELD, baseName + "$HashIterator$HashIteratorShuffler", "hashIter", "L" + baseName + "$HashIterator;");
+            mv.visitInsn(ACONST_NULL);
+            if("java/util/WeakHashMap".equals(baseName)){
+                mv.visitFieldInsn(PUTFIELD, baseName + "$HashIterator", "lastReturned", "L" + baseName + "$" + type + ";");
```

Comment:
Also, please add the source for WeakHashMap. It seems that the source for IdentityHashMap is also missing... please check what else is missing.